File: TDDMicroExercises/PHP/TelemetrySystem/TelemetryDiagnosticControls.php
<?php

namespace TDDMicroExercises\PHP\TelemetrySystem;

class TelemetryDiagnosticControls
{
	const DiagnosticChannelConnectionString = "*111#";

	private $_telemetryClient;
	private $_diagnosticInfo = '';

	public function __construct()
	{
		$this->_telemetryClient = new TelemetryClient();
	}

	public function getDiagnosticInfo()
	{
		return $this->_diagnosticInfo;
	}

	public function setDiagnosticInfo( $diagnosticInfo )
	{
		$this->_diagnosticInfo = $diagnosticInfo;
	}

	public function checkTransmission()
	{
		$this->_diagnosticInfo = '';
		$this->_telemetryClient->disconnect();

		$retryLeft = 3;
		while ( $this->_telemetryClient->getOnlineStatus() == false && $retryLeft > 0 ) 
		{
			$this->_telemetryClient->connect(
				TelemetryDiagnosticControls::DiagnosticChannelConnectionString
			);
			$retryLeft -= 1;
		}

		if($this->_telemetryClient->getOnlineStatus() == false)
		{
			throw new \Exception('Unable to connect.');
		}

		$this->_telemetryClient->send(TelemetryClient::DiagnosticMessage);
		$this->_diagnosticInfo = $this->_telemetryClient->receive();
	}
}

File: TDDMicroExercises/Java/TelemetrySystem/src/main/java/tddmicroexercises/telemetrysystem/TelemetryDiagnosticControls.java
package tddmicroexercises.telemetrysystem;

public class TelemetryDiagnosticControls
{
    private static final String DIAGNOSTIC_CHANNEL_CONNECTION_STRING = "*111#";

    private final TelemetryClient telemetryClient;
    private String diagnosticInfo = "";

    public TelemetryDiagnosticControls()
    {
        telemetryClient = new TelemetryClient();
    }

    public String getDiagnosticInfo()
    {
        return diagnosticInfo;
    }

    public void setDiagnosticInfo(String diagnosticInfo)
    {
        this.diagnosticInfo = diagnosticInfo;
    }

    public void checkTransmission() throws Exception
    {
        diagnosticInfo = "";

        telemetryClient.disconnect();

        int retryLeft = 3;
        while (!telemetryClient.getOnlineStatus() && retryLeft > 0)
        {
            telemetryClient.connect(DIAGNOSTIC_CHANNEL_CONNECTION_STRING);
            retryLeft -= 1;
        }

        if (!telemetryClient.getOnlineStatus())
        {
            throw new Exception("Unable to connect.");
        }

        telemetryClient.send(TelemetryClient.DIAGNOSTIC_MESSAGE);
        diagnosticInfo = telemetryClient.receive();
    }
}
